adapters/yieldmo/yieldmo.go

```diff
@@ -0,0 +1,141 @@
+package yieldmo
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type YieldmoAdapter struct {
+	endpoint string
+}
+
+type Ext struct {
+	PlacementId string `json:"placement_id"`
+}
+
+func (a *YieldmoAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+
+	adapterReq, errors := a.makeRequest(request)
+	if adapterReq != nil {
+		adapterRequests = append(adapterRequests, adapterReq)
+	}
+	errs = append(errs, errors...)
+
+	return adapterRequests, errors
+}
+
+func (a *YieldmoAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	// Make a copy as we don't want to change the original request
+	reqCopy := *request
+	if err := preprocess(&reqCopy); err != nil {
+		errs = append(errs, err)
+	}
+
+	// Last Step
+	reqJSON, err := json.Marshal(reqCopy)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+	}, errs
+}
+
+// Mutate the request to get it ready to send to yieldmo.
+func preprocess(request *openrtb.BidRequest) error {
+	for i := 0; i < len(request.Imp); i++ {
+		var imp = request.Imp[i]
+		var bidderExt adapters.ExtImpBidder
+
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		var yieldmoExt openrtb_ext.ExtImpYieldmo
+
+		if err := json.Unmarshal(bidderExt.Bidder, &yieldmoExt); err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		var impExt Ext
+		impExt.PlacementId = yieldmoExt.PlacementId
+
+		impExtJSON, err := json.Marshal(impExt)
+		if err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		imp.Ext = impExtJSON
+		request.Imp[i] = imp
```

Comment:
Given my previous comment, it should be safe to just `request.Imp[i].Ext = impExtJSON` as PBS has already insured you are working on your own private copy of `Imp`, saving some copy steps. I am happy to see care taken against generating race conditions by ensuring parallel routines do not try to modify the same data.

Reply:
Cleaned this up.